src/wallet/api.rs

```diff
@@ -1502,78 +1462,84 @@ impl Wallet {
-    }
-
-    /// Imports a multisig redeem script with a descriptor into the wallet.
-    fn import_multisig_redeemscript_descriptor(
-        &self,
-        pubkey1: &PublicKey,
-        pubkey2: &PublicKey,
-        address_label: &String,
-    ) -> Result<(), WalletError> {
+        let spk = redeemscript_to_scriptpubkey(redeemscript);
         let descriptor = self
             .rpc
-            .get_descriptor_info(&format!("wsh(sortedmulti(2,{},{}))", pubkey1, pubkey2))?
+            .get_descriptor_info(&format!("raw({:x})", spk))
+            .unwrap()
             .descriptor;
-        let result = self
-            .rpc
-            .import_multi(
-                &[ImportMultiRequest {
-                    timestamp: Timestamp::Now,
-                    descriptor: Some(&descriptor),
-                    watchonly: Some(true),
-                    label: Some(address_label),
-                    ..Default::default()
-                }],
-                Some(
-                    &(ImportMultiOptions {
-                        rescan: Some(false),
-                    }),
-                ),
-            )
-            .unwrap();
-        for r in result {
-            if !r.success {
-                log::warn!(target: "Wallet:import_addresses", "{:?}", r.warnings);
-                if let Some(e) = r.error {
-                    return Err(WalletError::Protocol(e.message));
-                }
-            }
-        }
-        Ok(())
+        self.import_descriptors(&[descriptor], Some(WATCH_ONLY_SWAPCOIN_LABEL.to_string()))
     }
 
-    /// Imports a redeem script into the wallet.
-    pub fn import_redeemscript(
-        &self,
-        redeemscript: &ScriptBuf,
-        address_label: &String,
-    ) -> Result<(), WalletError> {
-        let spk = redeemscript_to_scriptpubkey(redeemscript);
-        let result = self.rpc.import_multi(
-            &[ImportMultiRequest {
-                timestamp: Timestamp::Now,
-                script_pubkey: Some(ImportMultiRequestScriptPubkey::Script(&spk)),
-                redeem_script: Some(redeemscript),
-                watchonly: Some(true),
-                label: Some(address_label),
-                ..Default::default()
-            }],
-            Some(
-                &(ImportMultiOptions {
-                    rescan: Some(false),
-                }),
-            ),
-        )?;
-        for r in result {
-            if !r.success {
-                log::warn!(target: "Wallet:import_addresses", "{:?}", r.warnings);
-                if let Some(e) = r.error {
-                    return Err(WalletError::Protocol(e.message));
-                }
-            }
-        }
-        Ok(())
+    pub fn collect_descriptors_to_import(&self) -> Result<Vec<String>, WalletError> {
+        let mut descriptors_to_import = Vec::new();
+
+        descriptors_to_import.extend(self.get_unimported_wallet_desc()?);
```

Comment:
Does it make sense to remove this check now? Its two extra RPC calls at every sync time.

Reply:
The first call is required to get the descriptor with checksum and another one is for checking if the descriptor is imported or not. I am thinking of implementing the checksum method, and thus eliminating the need for 1st RPC call. Let's import every descriptor, and see if it reduces the time, thus eliminating the 2nd RPC call.